src/services/formatting/rules.ts

```diff
@@ -725,17 +725,7 @@ namespace ts.formatting {
     }
 
     function isEndOfDecoratorContextOnSameLine(context: FormattingContext): boolean {
-        return context.TokensAreOnSameLine() &&
-            hasDecorators(context.contextNode) &&
-            nodeIsInDecoratorContext(context.currentTokenParent) &&
-            !nodeIsInDecoratorContext(context.nextTokenParent);
-    }
-
-    function nodeIsInDecoratorContext(node: Node): boolean {
-        while (isExpressionNode(node)) {
-            node = node.parent;
-        }
-        return node.kind === SyntaxKind.Decorator;
+        return context.TokensAreOnSameLine() && hasDecorators(context.contextNode) && isDecorator(context.currentTokenParent);

```

Comment:
Wouldn't this falsely identify the following as the end of a decorator context:
```ts
class C {
  constructor(@Foo @Bar param: any) {}
           //  ^^^
}
```

If for some reason the current token is `Foo`, its parent will be a decorator, but the next token is *also* in a decorator context.

Reply:
@rbuckton Thanks for the review. I just thought [formattingDecorators](https://github.com/microsoft/TypeScript/blob/main/tests/cases/fourslash/formattingDecorators.ts) covered most cases. The rule was [added](https://github.com/microsoft/TypeScript/commit/eec39c2fc50fdeee05b385e8c83fd54992e03d87) without tests, so it's difficult to understand what cases it covers.